src/AutoRest.CSharp/Common/Output/Builders/BuilderHelpers.cs

```diff
@@ -220,6 +221,13 @@ public static CSharpType GetTypeFromExisting(ISymbol existingMember, CSharpType
             return PromoteNullabilityInformation(newType, defaultType);
         }
 
+        public static bool IsTypeFromExistingReadOnly(ISymbol existingMember) => existingMember switch
```

Comment:
should this be `IsReadOnlyFromExisting`?
We are getting if the "property/field" is readonly instead of the type.